test/functional/apm_logging_tests.js

```diff
@@ -0,0 +1,225 @@
+      spy.restore();
+      done(err);
+    };
+    runTestClient(this, '?monitor=command&monitorOut=stderr', cb);
+  });
+
+  it('log only sdam events', function(done) {
+    const cb = err => {
+      const logFile = fs.readFileSync(testLogFile, 'utf8');
+      const data = getFileData(logFile);
+
+      expect(data.numLines).to.equal(6);
+      expect(data.command).to.be.empty;
+      done(err);
+    };
+    runTestClient(this, '?monitor=sdam&monitorOut=apm_test_log', cb);
+  });
+
+  it('log only command events', function(done) {
+    const cb = err => {
+      const logFile = fs.readFileSync(testLogFile, 'utf8');
+      const data = getFileData(logFile);
+
+      expect(data.numLines).to.equal(2);
+      expect(data.sdam).to.be.empty;
+      done(err);
+    };
+    runTestClient(this, '?monitor=command&monitorOut=apm_test_log', cb);
+  });
+
+  // skip for now: this test depends on the new fixes in the core uri-parser
+  it.skip('log both command and sdam events using `monitor=command,sdam`', function(done) {
+    const cb = err => {
+      const logFile = fs.readFileSync(testLogFile, 'utf8');
+      const data = getFileData(logFile);
+
+      expect(data.numLines).to.equal(8);
+      expect(data.sdam).to.not.be.empty;
+      expect(data.command).to.not.be.empty;
+      done(err);
+    };
+    runTestClient(this, '?monitor=command,sdam&monitorOut=apm_test_log', cb);
+  });
+
+  it('log both command and sdam events using `monitor=all`', function(done) {
+    const cb = err => {
+      const logFile = fs.readFileSync(testLogFile, 'utf8');
+      const data = getFileData(logFile);
+
+      expect(data.numLines).to.equal(8);
+      expect(data.sdam).to.not.be.empty;
+      expect(data.command).to.not.be.empty;
+      done(err);
+    };
+    runTestClient(this, '?monitor=all&monitorOut=apm_test_log', cb);
+  });
+
+  it('should log correct event objects in the same order they were emitted', function(done) {
+    const cb = (err, events) => {
+      const logFile = fs.readFileSync(testLogFile, 'utf8');
+      const data = getFileData(logFile);
+
+      expect(events).to.deep.equal(data.eventStrings);
+      done(err);
+    };
+    runTestClient(this, '?monitor=all&monitorOut=apm_test_log', cb);
+  });
+
+  it('should call callback with error when log filename cannot be resolved', {
+    metadata: { requires: { node: '<10.0.0' } },
+    test: function(done) {
+      const configuration = this.configuration;
+      const url = configuration.url() + '?monitor=all&monitorOut=invalid:name';
+      const client = new MongoClient(url, { w: 1, useNewUrlParser: true });
+
+      client.connect(function(err) {
+        expect(err).to.exist;
+        console.log(err);
+        expect(err.name).to.equal('TypeError');
+        expect(err.message).to.equal(`Path must be a string. Received { invalid: 'name' }`);
+        done();
+      });
+    }
+  });
+
+  it('should call callback with error when log filename cannot be resolved', {
```

Comment:
seems like this can be merged with the previous test. These tests are basically testing node itself at this point (which error message is returned). I think it suffices to remove the node version requirement, and asserting that `expect(err).to.be.instanceOf(TypeError)`.